include/envoy/grpc/async_client.h

```diff
@@ -83,17 +90,38 @@ template <class ResponseType> class AsyncRequestCallbacks {
                          Tracing::Span& span) PURE;
 };
 
+// Templatized variant of AsyncRequestCallbacks.
+template <class ResponseType> class TypedAsyncRequestCallbacks : public AsyncRequestCallbacks {
```

Comment:
@mattklein123 this pattern might make sense also (in a separate PR) for the filter instantiation interface. These templatized adapters allow the boiler plate around having to specify a `createEmptyProto()` method and then later `dynamic_cast` to be handled by the compiler.

Reply:
That sounds great. Let's do it!